library/qs_source/src/std/diagnostics.qs

```diff
@@ -149,5 +149,25 @@ namespace Microsoft.Quantum.Diagnostics {
         areEqual
     }
 
-    export DumpMachine, DumpRegister, CheckZero, CheckAllZero, Fact, CheckOperationsAreEqual;
+    @Config(Unrestricted)
+    operation StartCountingOperation<'In, 'Out>(callable : 'In => 'Out) : Unit {
+        body intrinsic;
+    }
+
+    @Config(Unrestricted)
+    operation StopCountingOperation<'In, 'Out>(callable : 'In => 'Out) : Int {
```

Comment:
I went back and forth on whether to call these `StopCountingOperation` or `GetOperationCounts`, but to me the latter does not imply tracking stops while the former can imply the final count is returned. If there is a strong preference for "get" instead, then I think we'll need three APIs: start, get, and stop. I prefer to keep it to two and minimize the API surface, personally.